From 96f148226008ff7310a5ecc0da4248fe7a7fe24e Mon Sep 17 00:00:00 2001
From: Arun Khandavalli <arun.khandavalli@mentor.com>
Date: Tue, 19 May 2015 10:57:29 +0530
Subject: [PATCH] ls1021atwr: Enable support for SD write to NOR flash.

By default IFC is disabled in SD configuration for u-boot because of
which we cannot write into NOR flash from SD card. This is important
for recovery purposes of NOR flash.

Signed-off-by: Alison Wang <alison.wang@freescale.com>
Signed-off-by: Arun Khandavalli <arun.khandavalli@mentor.com>
---
 board/freescale/ls1021atwr/ls1021atwr.c |   10 +++++-----
 board/freescale/ls1021atwr/ls1_rcw.cfg  |   20 ++++++++++----------
 include/configs/ls1021atwr.h            |    7 +++----
 3 files changed, 18 insertions(+), 19 deletions(-)

diff --git a/board/freescale/ls1021atwr/ls1021atwr.c b/board/freescale/ls1021atwr/ls1021atwr.c
index 84f9ec6..e43c582 100644
--- a/board/freescale/ls1021atwr/ls1021atwr.c
+++ b/board/freescale/ls1021atwr/ls1021atwr.c
@@ -89,7 +89,7 @@ struct cpld_data {
 static void convert_serdes_mux(int type, int need_reset);
 
 
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 void cpld_show(void)
 {
 	struct cpld_data *cpld_data = (void *)(CONFIG_SYS_CPLD_BASE);
@@ -131,7 +131,7 @@ void cpld_show(void)
 int checkboard(void)
 {
 	puts("Board: LS1021ATWR\n");
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 	cpld_show();
 #endif
 
@@ -382,7 +382,7 @@ int arch_early_init_r(void)
 	struct ccsr_gur __iomem *gur = (void *)(CONFIG_SYS_FSL_GUTS_ADDR);
 	u32 protocol = in_be32(&gur->rcwsr[4]) & RCWSR4_SRDS1_PRTCL_MASK;
 
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 #ifndef CONFIG_SYS_FSL_NO_SERDES
 	/* needs to be in ram since code uses global static vars */
 	/* add LC 05/14/2014 */
@@ -674,7 +674,7 @@ void config_mdio_mux(int addr)
 	return;
 }
 
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 static void convert_flash_bank(char bank)
 {
 	struct cpld_data *cpld_data = (void *)(CONFIG_SYS_CPLD_BASE);
@@ -798,7 +798,7 @@ void board_print_spi_device(void)
 	printf("S25FL064  is on spi bus 2 cs 0\n");
 }
 
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 static void convert_serdes_mux(int type, int need_reset)
 {
 	char current_serdes;
diff --git a/board/freescale/ls1021atwr/ls1_rcw.cfg b/board/freescale/ls1021atwr/ls1_rcw.cfg
index e210f6f..9354359 100644
--- a/board/freescale/ls1021atwr/ls1_rcw.cfg
+++ b/board/freescale/ls1021atwr/ls1_rcw.cfg
@@ -1,14 +1,14 @@
 #PBL preamble and RCW header
 aa55aa55 01ee0100
 # serdes protocol
-#first try
-#0608000a 00000000 00000000 00000000
-#00000000 00404000 60025a00 21042000
-#00200000 00000000 00000000 01038000
-#00000000 001b1200 00000000 00000000
-
-#disable IFC,enable the QSP and DSPI
+#enable IFC, disable QSPI and DSPI
 0608000a 00000000 00000000 00000000
-20000000 00407900 60025a00 21046000
-00000000 00000000 00000000 01038000
-20024800 881b1540 00000000 00000000
+20000000 00407900 60040a00 21046000
+00000000 00000000 00000000 00038000
+00080000 881b7340 00000000 00000000
+
+#disable IFC, enable QSPI and DSPI
+#0608000a 00000000 00000000 00000000
+#20000000 00407900 60040a00 21046000
+#00000000 00000000 00000000 00038000
+#20084800 881b7340 00000000 00000000
diff --git a/include/configs/ls1021atwr.h b/include/configs/ls1021atwr.h
index 0c58e76..2859c4f 100644
--- a/include/configs/ls1021atwr.h
+++ b/include/configs/ls1021atwr.h
@@ -135,7 +135,6 @@ unsigned long get_board_ddr_clk(void);
 #define CONFIG_SPL_BSS_MAX_SIZE		0x80000
 #define CONFIG_SYS_MONITOR_LEN		0x80000
 #define CONFIG_SYS_SPL_ASSUME_UBOOT_NIH
-#define CONFIG_SYS_NO_FLASH		/*For QSPI&DSPI verificaton*/
 #endif
 
 #define CONFIG_SYS_HAS_SERDES
@@ -177,12 +176,12 @@ unsigned long get_board_ddr_clk(void);
 #define CONFIG_DOS_PARTITION
 #endif	/* CONFIG_PCI */
 
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_NAND) && !defined(CONFIG_SPI)
+#if !defined(CONFIG_NAND) && !defined(CONFIG_SPI)
 #define CONFIG_U_QE
 #endif
 
 /* IFC Definitions */
-#if !defined(CONFIG_SDCARD) && !defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 #define CONFIG_FSL_IFC
 #define CONFIG_SYS_FLASH_BASE		0x60000000
 #define CONFIG_SYS_FLASH_BASE_PHYS	CONFIG_SYS_FLASH_BASE
@@ -290,7 +289,7 @@ unsigned long get_board_ddr_clk(void);
 
 #define CONFIG_BAUDRATE			115200
 
-#if defined(CONFIG_SDCARD) || defined(CONFIG_SPI)
+#ifndef CONFIG_SPI
 #undef CONFIG_CMD_IMLS
 #else
 #define CONFIG_CMD_IMLS
-- 
1.7.9.5

